tests/Unit/ApparentHorizons/StrahlkorperGrTestHelpers.cpp

```diff
@@ -103,6 +106,1883 @@ Scalar<DataType> horizon_radius(const std::array<DataType, 2>& theta_phi,
 
   return Scalar<DataType>{sqrt(radius_squared)};
 }
+
+template <typename DataType>
+Scalar<DataType> horizon_ricci_scalar(
+    const DataType& theta_points, const double mass,
+    const double dimensionless_spin_z) noexcept {
+  auto ricci_scalar = make_with_value<Scalar<DataType>>(theta_points, 0.0);
+
+  // Create versions of these variables with shorter names.
+  // Mathematica-generated code to come.
+  const double a = dimensionless_spin_z * mass;
+  const auto& theta = theta_points;
+
+  // Begin Mathematica-generated code
+  ricci_scalar.get() =
+      ((-pow(a, 2) + 2 * mass * (mass + sqrt(-pow(a, 2) + pow(mass, 2))) +
```

Comment:
in emacs, find and replace (mass + sqrt(-pow(a, 2) + pow(mass, 2)) with horizon_radius (verify that is the correct name) and then see if it simplifies further...

Reply:
Doing: `r_plus = (mass + sqrt(-pow(a, 2) + pow(mass, 2))` saves 700 lines...

Reply:
Hi Geoffrey,

See Eq. (119) of https://arxiv.org/pdf/0706.0622.pdf which is  Eq. (23) of a paper by Smarr (https://journals-aps-org.proxy.library.cornell.edu/prd/pdf/10.1103/PhysRevD.7.289)  which gives the 2 Ricci scalar on the horizon in Boyer–Lindquist coordinates.

You can then use the transformation from Boyer-Lindquist (r, \theta)  to Kerr-Schild (x, y, z)  to see that

r^2 + a^2 sin^2 \theta = x^2 + y^2 + z^2

Then a simple function taking `Kerr::horizon_radius` as input instead of `theta_points` can be written as you can use the previous formula to substitute

a^2 cos^2 \theta = r_plus^2 - horizon_radius^2  + a^2

into the Smarr formula.  Also, I think the resulting function should be valid for arbitrary spin direction.   I have tested that it agrees with the Mathematica generated function for spins in the z-direction.